### DISCUSSION TITLE
Suggestion dissappears before being able to accept it, mostly where the suggestion is needed on the current line

### DISCUSSION THREAD

**Discussion by RunRobert** [community/community#123942] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

When using Github copilot in Visual studio 2022 and VB.NET, a lot of the time GitHub Copilot only flashes it's suggestion, but too fast for you to tab into it. Sometimes I get to the last 2 characters before the suggestion is displayed to be selected. Why does it only flash the response briefly, and how do I change this behavior? It is not because I continue to type. I stop typing, wait for the suggestion, and then tab to select it. But the suggestion disappears before I can tab into it, forcing me to write the code manually. Why does it just flash the suggestion? It works fine if the code is on a new line. This appears to be an unhandled bug within Copilot. 

To be clear, I type a character, I stop typing, after a brief period a suggestion comes up, but then goes straight away without me typing anything. Tab just acts as a tab. I type another character. I stop. The suggestion appears for a very short period and then is gone. Sometime it only stays there when I am right at the end of typing the line for myself. This is very frustrating.

There is no issue with this when I type a comment, and the code suggestion appears on a new line. I can tab and it accepts the code.

**Comment by davidklempfner1** [comment]
I have this same issue. Very frustrating.

**Comment by ayim** [comment, marked as answer]
Sorry to hear it's been frustrating, I wonder if we have a race condition here.

I've filed this bug report - if you have any screen recordings (Win + G makes this really easy) and Visual Studio version info you could add to that, would be really helpful.

https://developercommunity.visualstudio.com/t/VBNET-Copilot-suggestion-disappears-bef/10688824

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Comment by gundermanc** [comment]
Hey folks, I am actively investigating this issue. Can you please share any videos of the repro if you have them or repro steps to aid in in the investigation?

[ScreenToGIF](https://www.screentogif.com/downloads) is another great tool for doing these sorts of captures.

**Reply by peteringemann** [reply]
Anything new? This bug is present in all my VS installations and it is rather annoying.

**Reply by gundermanc** [reply]
Hi @peteringemann, depending on the specific repro steps you may be encountering one of two types of issues with gray text completions.

- **ReSharper or Extension Incompatibility** - JetBrains ReSharper Extension for Visual Studio disables the VS Editor completion list control in C# and replaces it with its own. This specific behavior is currently incompatible with the gray text UI in the VS Editor. If you have ReSharper installed, I recommend disabling the ReSharper IntelliSense list via the following option when using Copilot Completions. If you would like to see [support for ReSharper completions added to the gray text UI, you can vote for it here](https://developercommunity.visualstudio.com/t/GitHub-Copilot-Completions-in-Visual-Stu/10688821).

![image](https://github.com/user-attachments/assets/7f475d6c-1a61-4177-9bab-0a3ab268ff02)

- **Gray text UI bugs** - [In July I reported investigating issues](https://github.com/orgs/community/discussions/123942#discussioncomment-10153303) with disappearing gray text. We have since shipped several fixes, all of which should be in Visual Studio 17.12 and newer. If you can still repro the flashing or disappearing issue and ReSharper Completions is not installed or is disabled, you have probably discovered a new issue.

If you are still encountering an issue and to maximize the possibility of an issue being fixed, please:
- Use _Help > Send Feedback > Report a problem_ for each specific instance of the issue, provide repro steps, and use tools like ScreenToGIF to capture a GIF of the repro. Issues and repros can be very scenario specific and VS Feedback tool and a visual indication of what's happening is essential to helping engineering team diagnose issues.
- When filing GitHub issues or questions, please also report via VS Feedback tool and link to that issue.

**Reply by RobsBigNose** [reply]
I don't have Resharper installed. So has to be a new bug. Maybe specific to VB.

**Reply by gundermanc** [reply]
Do you have a minimal repro project you can share?

**Reply by RobsBigNose** [reply]
This happens on any project I work on. It also doesn't do this all the time.

**Comment by RunRobert** [comment]
Here is a video of some of the times this has happened. Note that I am typing slow so that it can guess. You will see it flash the answer, and dissapear even though I am not actively typing at that moment. 

Uploading GitHub Copilot Issues.mov…

**Reply by ayim** [reply]
Hey @RunRobert, looks like the video didn't upload - mind trying again?

**Comment by RobsBigNose** [comment]
Here is the file again

https://github.com/user-attachments/assets/c5dede92-7fb3-43cf-8c90-1d985ff93ea4

**Reply by gundermanc** [reply]
@RobsBigNose I'm not able to repro this in 17.12 in a basic VB ASPX app. Are you using 17.12? If so, do you have a minimal repro project that you can share?

**Reply by RobsBigNose** [reply]
I am running 17.7.6. I will upgrade sometime today and see if it makes any difference.

**Reply by gundermanc** [reply]
We fixed several issues in 17.12 so that will very likely be an improvement. One in particular was causing issues with gray text being dismissed in VB files specifically.
